internal/handler/user.go

```diff
@@ -0,0 +1,60 @@
+package handler
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/dilyara4949/flight-booking-api/internal/domain"
+	"github.com/dilyara4949/flight-booking-api/internal/handler/request"
+	"github.com/dilyara4949/flight-booking-api/internal/handler/response"
+	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
+)
+
+type UserService interface {
+	CreateUser(ctx context.Context, signup request.Signup, password string) (domain.User, error)
+	GetUser(ctx context.Context, id uuid.UUID) (domain.User, error)
+	UpdateUser(ctx context.Context, req request.UpdateUser, userID uuid.UUID) (domain.User, error)
+}
+
+func UpdateUserHandler(userService UserService) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		var req request.UpdateUser
+
+		err := c.ShouldBind(&req)
+		if err != nil {
+			c.JSON(http.StatusBadRequest, response.Error{Error: "error at binding request body"})
+			return
+		}
+
+		if req.Email == "" || req.Phone == "" {
+			c.JSON(http.StatusBadRequest, response.Error{Error: "fields cannot be empty"})
+			return
+		}
+
+		userID, err := uuid.Parse(c.Param("userId"))
```

Comment:
You should consider that user_id might be also an optional request parameter. In that case we should check if the user have authorization to edit another user data (e.g. have an admin role). In this case  we take the parameter from request instead of that in token